### PR TITLE
Gate proxy rewrite

### PR DESCRIPTION
**Pull request by rianfowler**
## Description
- add gating behavior as described in ticket

## Testing done
- verify locally with proxy 

## Screenshot preview.va.gov/health 
header always activates regardless of cooke value 
screen shot 2018-10-18 at 15 46 33

## Screenshot www.cem.va.gov 
header can be toggled by adding or removing cookie `proxyRewrite=true;`
(broken images are caused by absolute paths for images- will be resolved in another ticket)
screen shot 2018-10-18 at 15 47 06
screen shot 2018-10-18 at 15 46 52

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/proxy-rewrite/proxy-rewrite-entry.jsx [hunk 1]
```diff
@@ -108,4 +101,54 @@ document.addEventListener('DOMContentLoaded', _e => {
...
+function shouldActivateInjectedAssets(whitelistItem, proxyRewriteCookieValue) {
+  if (whitelistItem.cookieOnly) {
+    return proxyRewriteCookieValue;
+  }
+
+  return !!whitelistItem;
+}
+
+if (
+  shouldActivateInjectedAssets(
```
Comment (rianfowler) [inline src/applications/proxy-rewrite/proxy-rewrite-entry.jsx]: most of the changes are just moving code around into named functions. This is where the new gating behavior begins.

File: src/applications/proxy-rewrite/proxy-rewrite-whitelist.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        "cookieOnly": false
+    },
+    {
+        "hostname": "staging.va.gov",
+        "pathnameBeginning": "/health/",
+        "cookieOnly": false
+    },
+    {
+        "hostname": "preview.va.gov",
+        "pathnameBeginning": "/health/",
```
Comment (rianfowler) [inline src/applications/proxy-rewrite/proxy-rewrite-whitelist.json]: this fulfills the requirements for now- will probably move away from json and store regex in code if we need to accommodate more complicated patterns.

**Review by cehsu — APPROVED** [review]
LGTM. Should any unit tests be written or updated?

File: src/applications/proxy-rewrite/proxy-rewrite-whitelist.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        "cookieOnly": false
+    },
+    {
+        "hostname": "www.cem.va.gov",
```
[the remark this answers is not in the document]
Comment (rianfowler) [reply src/applications/proxy-rewrite/proxy-rewrite-whitelist.json]: the sites are https://benefits.va.gov and https://www.cem.va.gov. http://cem.va.gov gets redirected. no idea why it's this way.

**Pull request merged**